pkg/web/mobileServicesHandler.go

```diff
@@ -116,43 +116,69 @@ func (msh *MobileServiceHandler) Create(rw http.ResponseWriter, req *http.Reques
 	rw.WriteHeader(http.StatusCreated)
 }
 
-// Configure configures components binding TODO NEEDS A REFACTOR
+// Configure configures components binding
 func (msh *MobileServiceHandler) Configure(rw http.ResponseWriter, req *http.Request) {
 	rw.Header().Set("Content-Type", "application/json")
 	token := headers.DefaultTokenRetriever(req.Header)
 
-	decoder := json.NewDecoder(req.Body)
-	var conf mobile.ServiceIntegration
-	err := decoder.Decode(&conf)
-	if err != nil {
-		handleCommonErrorCases(err, rw, msh.logger)
+	params := mux.Vars(req)
+	component := strings.ToLower(params["component"])
+	secret := strings.ToLower(params["secret"])
+	if len(component) == 0 {
+		handleCommonErrorCases(errors.New("web.msh.Configure -> provided component must not be empty"), rw, msh.logger)
+		return
+	}
+	if len(secret) == 0 {
```

Comment:
this is ok just wondering why you went that route instead of just ```if secret == ""```

Reply:
hmm, no real reason, it just felt better this way!